core/src/main/java/org/apache/iceberg/encryption/EnvelopeMetadata.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.encryption;
+
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.Objects;
+import org.apache.iceberg.relocated.com.google.common.base.MoreObjects;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+
+/**
+ * Envelope encryption metadata used to encrypt and decrypt data.
+ * Information except plaintext KEK and DEK can be serialized and stored.
+ */
+public class EnvelopeMetadata implements EncryptionKeyMetadata {
+
+  private final String kekId;
+  private final String wrappedDek;
+  private final byte[] iv;
+  private final EncryptionAlgorithm algorithm;
+
+  // fields that are expected to be updated later
+  private byte[] kek;
+  private byte[] dek;
+
+  public EnvelopeMetadata(
```

Comment:
Nit: they can be in the same line.

Reply:
seems  this is an accepted pattern for constructors with many parameters; also, looks cleaner as a column

Reply:
Iceberg style is supposed to be the block based version, everything on the same line with the first overflowing arg matching up with the first arg. I also prefer the column style but it's not what we use in Iceberg

Reply:
Sure, will change / add accordingly